packages/button/src/react/index.tsx

```diff
@@ -128,29 +128,33 @@ interface ButtonStatics {
   iconAligns: typeof vars.iconAligns
   sizes: typeof vars.sizes
 }
-
 interface BaseButtonProps {
   appearance?: ValueOf<typeof vars.appearances>
+  // TODO: rm? dup of html attr
   disabled?: boolean
   icon?: React.ReactNode
   iconAlign?: ValueOf<typeof vars.iconAligns>
   loading?: boolean
   size?: ValueOf<typeof vars.sizes>
 }
-
-type ButtonTagProps = BaseButtonProps &
-  React.ButtonHTMLAttributes<HTMLButtonElement>
-type AnchorTagProps = BaseButtonProps &
-  React.AnchorHTMLAttributes<HTMLAnchorElement>
-
-type ButtonProps = ButtonTagProps | AnchorTagProps
-
-interface ButtonComponent
-  extends RefForwardingComponent<
-    ButtonProps,
-    HTMLAnchorElement | HTMLButtonElement,
-    ButtonStatics
-  > {}
+type AnchorEl = JSX.IntrinsicElements['a']
+interface ButtonAnchorProps extends BaseButtonProps, AnchorEl {
+  href: string
+}
+type ButtonEl = JSX.IntrinsicElements['button']
+interface ButtonButtonProps extends BaseButtonProps, ButtonEl {
+  href?: undefined
+}
+type ButtonProps = ButtonAnchorProps | ButtonButtonProps
+
+type ButtonComponent = RefForwardingComponent<
+  ButtonProps,
+  HTMLAnchorElement | HTMLButtonElement,
+  ButtonStatics
+> & {
+  (props: ButtonAnchorProps, ref?: RefObject<HTMLAnchorElement>): JSX.Element
+  (props: ButtonButtonProps, ref?: RefObject<HTMLButtonElement>): JSX.Element
+}
```

Comment:
the linter doesn't seem to like this style of overriding

Reply:
Huh, seems to have made it through the last ci lint ok.  What is flagging for you?

Reply:
ci was erroring out. if that passed i'm not concerned

Reply:
Maybe I checked too fast.  We'll see this time around.  CI looked like a test failure to me, which was in Banner, tsc-related.  Maybe there was something else in there too...